crates/ggml/src/tensor.rs

```diff
@@ -9,6 +9,12 @@ pub struct Tensor {
     pub(crate) ctx: Weak<NonNull<sys::ggml_context>>,
 }
 
+impl Clone for Tensor {
```

Comment:
I was considering doing this myself, but decided not to do it because the semantics are unclear - that is, you are not actually cloning the tensor data, just a handle to it, which means you can't take your newly cloned `Tensor` away from the context that owns it.

Maybe we should rename this to `TensorHandle` or something?

Reply:
If we decide not to implement `Clone` for `Tensor` this way, we can still achieve the same by manually implementing `Clone` for each model's `Layer` by calling `.share()` on each of its tensors.

That would be more verbose on each model implementation, but would make it clear that `Tensor` doesn't implement `Clone`.

What would you suggest renaming to `TensorHandle`?